spec/spec.md

```diff
@@ -836,7 +877,298 @@ processing-related rules above:
+        }
+        ```
+      - `greater-than or equal-to` - Use the `minimum` descriptor. For
+        example, to request a proof that an attribute is greater than or
+        equal to 18, use the following as the value of the `filter`
+        object:
+        ```json
+        {
+          "type": "number",
+          "minimum": 18,
+        }
+        ```
+      - `less-than or equal-to` - Use the `maximum` descriptor. For
+        example, to request a proof that an attribute is less than or
+        equal to 65536, use the following as the value of the `filter`
+        object:
+        ```json
+        {
+          "type": "number",
+          "maximum": 65536,
+        }
+        ```
+  - to express the following equality proofs, use the JSON Schema
+    `const` descriptor:
+      - `equal-to` - Use the `const` descriptor. For example to
+        request proof that an attribute has the value "Chad", use the
+        following as the value of the `filter` object:
+        ```json
+        {
+          "const": "Chad"
+        }
+        ```
+      - `not equal-to` - Use the `const` descriptor with the `not`
+        operator. For example, to request proof that an attribute does
+        not have the value "Karen", use the following as the value of
+        the `filter` object:
+        ```json
+        {
+          "not": {
+            "const": "Karen"
+          }
+        }
+        ```
+  - to express set-membership proofs, use the JSON Schema `enum`
+    descriptor:
+      - `in-set` - Use the `enum` descriptor. For example, to
+        request proof that an attribute is contained in the set of
+        rainbow colors, use the following as the value of the `filter`
+        object:
+        ```json
+        {
+          "type": "string",
+          "enum": ["red", "yellow", "blue"]
+        }
+        ```
+      - `not-in-set` - Use the `enum` descriptor with the `not`
+        operator. For example, to request proof that an attribute is
+        not contained in the set of primary colors, use the following
+        as the value of the `filter` object:
+        ```json
+        {
+          "not": {
+            "enum": ["red", "yellow", "blue"]
+          }
+        }
+        ```
+
+At this time, additional predicate operations are not supported.
+
+
+## Relational Constraint Feature
+
+The Relational Constraint [[ref:Feature]] extends the [[ref:Input Descriptor Object]]
+`constraints` object with additional properties.
+
+When using this [[ref:Feature]]:
+- The _constraints object_ ****MAY**** contain a `subject_is_issuer`
+  property. If present, its value ****MUST**** be one of the following
+  strings:
+    - `required` - This indicates that the processing entity ****MUST****
+      submit a response that has been _self-attested_, i.e., the
+      [[ref:Claim]] used in the presentation was 'issued' by the
+      [[Ref:Subject]] of the [[ref:Claim]].
+    - `preferred` - This indicates that it is ****RECOMMENDED**** that the
```

Comment:
What is the use case for a preferred subject_is_issuer from a Verifier perspective? If I don't care I would not put the constraint in to begin with as it is a MAY.

I do see a use case for the inverse of required, basically stating that a subject is not the issuer.

I guess this comment is applicable to all relational constraints.

Reply:
Yeah, but then you are expressing it at a per VC level anyway right? I guess it would make it a bit more expressive, but if I create a definition that simply leaves out the `preferred` for the relational constraint, I would have exactly the same constraint as it is only expressing purpose/intent.

Reply:
iirc, the original set of possible values was 'required' and 'allowed', but it was changed to 'preferred' to match the other properties.
'allowed' makes much more sense as a possible option, i.e., I need to know your address and it can come some issuer or from yourself.

Reply:
@brentzundel please create an issue